admin/onpage/class-onpage.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Handle the request for getting the onpage status
+ */
+class WPSEO_OnPage {
+
+	/**
+	 * The name of the user meta key for storing the dismissed status.
+	 */
+	const USERMETAVALUE = 'wpseo_dismiss_onpage';
+
+	/**
+	 * @var WPSEO_OnPage_Option The OnPage.org option class.
+	 */
+	private $onpage_option;
+
+	/**
+	 * Constructing the object
+	 */
+	public function __construct() {
+		// Only when AJAX isn't loaded.
+		if ( ! ( defined( 'DOING_AJAX' ) && DOING_AJAX === true ) ) {
+			$this->onpage_option = new WPSEO_OnPage_Option();
+
+			$this->set_hooks();
+			$this->catch_redo_listener();
+		}
+	}
+
+	/**
+	 * Adding a weekly schedule to the schedules array
+	 *
+	 * @param array $schedules Array with schedules.
+	 *
+	 * @return array
+	 */
+	public function add_weekly_schedule( array $schedules ) {
+		$schedules['weekly'] = array( 'interval' => WEEK_IN_SECONDS, 'display' => __( 'Once Weekly' ) );
+
+		return $schedules;
+	}
+
+	/**
+	 * Fetching the data from onpage.
+	 *
+	 * @return bool
+	 */
+	public function fetch_from_onpage() {
+		if ( $this->onpage_option->can_fetch() ) {
+			// Updates the timestamp in the option.
+			$this->onpage_option->set( 'last_fetch', time() );
+
+			// The currently indexability status.
+			$old_status = $this->onpage_option->get( 'status' );
```

Comment:
If there are only a limited amount of options it would be useful to add a `get_status` function to `WPSEO_OnPage_Option`. If there are a ton of options we could opt for `get( WPSEO_OnPage_Option::STATUS )`.